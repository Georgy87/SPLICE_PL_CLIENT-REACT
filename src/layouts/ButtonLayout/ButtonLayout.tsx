import React from 'react';
import { Button } from 'antd';

import styles from './ButtonLayout.module.scss';

export const ButtonLayout: React.FC = ({ children }) => {
	return (
		<div className={styles.btnContainer}>
			<Button>{children}</Button>
		</div>
	);
};
