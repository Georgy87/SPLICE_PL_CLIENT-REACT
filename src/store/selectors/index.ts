export * from './userSelectors';
export * from './packsSelectors';
export * from './samplesSelectors';