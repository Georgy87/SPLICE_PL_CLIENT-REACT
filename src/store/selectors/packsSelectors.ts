import { RootState } from '../types';

export const selectPackMain = (state: RootState) => state.packs;

export const selectPacks = (state: RootState) => selectPackMain(state).packs;
export const selectSamples = (state: RootState) => selectPackMain(state).packProfile?.samples;
export const selectPackProfile = (state: RootState) => selectPackMain(state).packProfile;
export const selectUserPacks = (state: RootState) => selectPackMain(state).userPacks;
