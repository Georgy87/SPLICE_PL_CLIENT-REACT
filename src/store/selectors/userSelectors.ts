import { RootState } from '../types';

export const selectUserMain = (state: RootState) => state.user;

export const selectAuth = (state: RootState) => selectUserMain(state).isAuth;
export const selectUser = (state: RootState) => selectUserMain(state).user;
export const selectLikedSamples = (state: RootState) => selectUserMain(state).samples;
export const selectErrorMessage = (state: RootState) => selectUserMain(state).errorMessage;

