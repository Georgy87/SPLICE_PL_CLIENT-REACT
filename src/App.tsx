import React, { useEffect, useRef } from 'react';
import { Route, Switch, useHistory } from 'react-router';
import { useDispatch, useSelector } from 'react-redux';

import { PacksPage } from './pages/PacksPage/PacksPage';
import { Navbar } from './components/Navbar/Navbar';
import { UserProfilePage } from './pages/UserProfilePage/UserProfilePage';
import { CreatePackPage } from './pages/CreatePackPage/CreatePackPage';
import { ProfilePackPage } from './pages/ProfilePackPage/ProfilePackPage';
import { LoginPage } from './pages/LoginPage/LoginPage';
import { PlayerContextProvider } from './context/PlayerContextProvider/PlayerContextProvider';
import { RegistrationPage } from './pages/RegistrationPage/RegistrationPage';
import { fetchAuth } from './store/slices/user/userSlice';
import { selectUser } from './store/selectors/userSelectors';
import { UserPacksPage } from './pages/UserPacksPage/UserPacksPage';
import { AvatarEditorPage } from './pages/AvatarEditorPage/AvatarEditorPage';
import { SequencerPage } from './pages/SequencerPage/SequencerPage';

import styles from './App.module.scss';

export const App: React.FC = () => {
	const user = useSelector(selectUser);

	const history = useHistory();
	const dispatch = useDispatch();

	useEffect(() => {
		// if (!user) {
		// 	history.push('login');
		// }

		dispatch(fetchAuth());
	}, []);

	return (
		<PlayerContextProvider>
			<div className={styles.appContainer}>
				<div className={styles.navbarBlock}>
					<Navbar />
				</div>
				<div className={styles.pagesBlock}>
					<Switch>
						<Route path='/login' component={LoginPage} exact />
						<Route path='/registration' component={RegistrationPage} exact />
						<Route
							path={'/'}
							component={() => <PacksPage pageName={'main-packs'} />}
							exact
						/>
						<Route path='/profile-pack/:packId?' component={ProfilePackPage} exact />
						<Route path='/profile' component={UserProfilePage} exact />
						<Route path='/profile/create' component={CreatePackPage} exact />
						<Route path='/profile/packs' component={UserPacksPage} exact />
						<Route path='/profile/avatar' component={AvatarEditorPage} exact />
						<Route path='/sequencer' component={SequencerPage} exact />
					</Switch>
				</div>
			</div>
		</PlayerContextProvider>
	);
};
