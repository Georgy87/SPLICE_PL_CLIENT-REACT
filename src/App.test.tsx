import fetchMock from 'jest-fetch-mock';
import { cleanup, fireEvent, render, screen, waitFor, act } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import * as actions from './store/slices/user/actions';
import userEvent from '@testing-library/user-event';

// import { act } from 'react-dom/test-utils';

import { App } from './App';
// import { LoginPage } from './pages';
import { userSlice } from './store/slices/user/userSlice';
import { MOCK_INITIAL_STATE, renderWithRouter, renderWithStore } from './__tests__/testUtils';
import { mockedServerResponse } from './__mocks__/api.mocks';
import { selectAuth } from './store/selectors/userSelectors';
import { selectPacks } from './store/selectors/packsSelectors';
import { ButtonLayout } from './layouts/ButtonLayout/ButtonLayout';

// jest.mock('react-redux');

// describe('App', () => {
//     const user = {
//         password: '345345',
//         avatar: '',
//         confirm_hash: 'hsdfgsdfg',
//         createdAt: '534',
//         email: 'gosga@jmail',
//         fullname: '',
//         updatedAt: '',
//         _id: '',
//     };

//     const initState = {
//         ...MOCK_INITIAL_STATE,
//         user: userSlice.reducer(undefined, setAuth({ user, token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjYxOGViYjVhNjI5M2MzMGY0MTU2ODAyYSIsImlhdCI6MTY3MDg3MDA5MiwiZXhwIjoxNjczNDYyMDkyfQ.twvF3KBaNnVUXO_lDcYsU5e4ipG7dY978dXqFXQXTpY" })),
//     };

//     it('App snapshot', async () => {
//         const { store } = renderWithStore(<App />, initState);

//         // await waitFor(() =>
//         //     expect(store.getActions().map(({ type }) => type)).toEqual([
//         //         'packs/setLoading',
//         //         'packs/setDefaultPackState',
//         //         'packs/setPacks',
//         //     ])
//         // );
//         // console.log(store.getActions(), store.getState());
//     });

// });
// const mockDispatch = jest.fn();
// const mockSelector = jest.fn();

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    //  useAppDispatch: () => mockDispatch,
    //  useSelector: () => mockSelector,
}));

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
const mockOnSubmit = jest.fn();
describe('login', () => {
    beforeEach(() => {
        fetchMock.resetMocks();
    });

    it('logineeee', async () => {
        // jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(false);
        // mockedDispatch.mockReturnValue(jest.fn());

        // const result = renderWithStore(<LoginPage />);

        // expect(result).toMatchSnapshot();
    });

    it('dispatch action', async () => {
        // const dispatch = jest.fn();

        // mockedDispatch.mockReturnValue(dispatch);
        const mockedFetchLogin = jest.spyOn(actions, 'fetchLogin');

        // const { result } = renderWithStore(<LoginPage />);

        // await act(async () => {
        //     fireEvent.change(screen.getByTestId('login_input'), { target: { value: 'test@mail.ru' } });
        //     fireEvent.change(screen.getByTestId('password_input'), { target: { value: 'TestPassword123_' } });
        // });

        // fireEvent.change(screen.getByTestId('login_input'), {
        //     target: { value: 'test@mail.ru' },
        // });

        // fireEvent.change(screen.getByTestId('password_input'), {
        //     target: { value: 'TestPassword123_' },
        // });

        // expect(result.getByTestId('submit_button')).toBeTruthy();

        // fireEvent.submit(result.getByTestId('submit_button'));
        // await act(async () => {
        //     fireEvent.submit(result.getByTestId('submit_button'));
        // });

        // await act(async () => {
        //     fireEvent.submit(result.getByRole('button'));
        // });
        
        userEvent.type(screen.getByPlaceholderText(/Email/i), 'test@mail.ru');
        userEvent.type(screen.getByPlaceholderText(/Password/i), 'TestPassword123_');
        userEvent.click(screen.getByText(/submit/i));

        // //@ts-ignore
        // expect(screen.getByPlaceholderText('Email').value).toBe('test@mail.ru');
        // //@ts-ignore
        // expect(screen.getByPlaceholderText('Password').value).toBe('TestPassword123_');

        // expect(mockLogin).toHaveBeenCalledTimes(1);

        // expect(mockedFetchLogin).toHaveBeenCalledTimes(1);
        // expect(mockOnSubmit).toHaveBeenCalled();

        await waitFor(() => {
            expect(mockOnSubmit).toHaveBeenCalledTimes(1);
        });
    });
});
