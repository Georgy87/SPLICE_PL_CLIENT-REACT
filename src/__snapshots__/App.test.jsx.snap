// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="packs-page"
      >
        <div
          class="player"
        >
          <video
            autoplay=""
            class="playerVideo"
            loop=""
            src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
          />
          <div
            class="initialInformation"
          >
            <h1>
              More than a sample library
            </h1>
            <p>
              Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
            </p>
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <label>
            Sounds
          </label>
          <div
            class="search"
          >
            <svg
              class="searchIcon"
              viewBox="0 0 512 512"
              width="95"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              class="searchInput"
              placeholder=""
              type="text"
              value=""
            />
            <svg
              class="searchClose"
              viewBox="0 0 352 512"
              width="100"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
            <svg
              class="searchDelete"
              viewBox="0 0 640 512"
              width="100"
            >
              <path
                d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
              />
            </svg>
          </div>
        </div>
        <div
          class="root"
        >
          <div
            class="loaderWrapper"
          >
            <div
              class="loader"
            >
              <div
                class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="loaderPagination"
          />
        </div>
      </div>
      <div
        class="appContainer"
      >
        <div
          class="navbarBlock"
        >
          <nav
            class="navbar notAuth"
          >
            <a
              class="navbarLogo"
              data-testid="packs-link"
              href="/"
            >
              <div
                class="iconContainer"
              >
                <span
                  aria-label="customer-service"
                  class="anticon anticon-customer-service"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="customer-service"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 128c-212.1 0-384 171.9-384 384v360c0 13.3 10.7 24 24 24h184c35.3 0 64-28.7 64-64V624c0-35.3-28.7-64-64-64H200v-48c0-172.3 139.7-312 312-312s312 139.7 312 312v48H688c-35.3 0-64 28.7-64 64v208c0 35.3 28.7 64 64 64h184c13.3 0 24-10.7 24-24V512c0-212.1-171.9-384-384-384zM328 632v192H200V632h128zm496 192H696V632h128v192z"
                    />
                  </svg>
                </span>
              </div>
              SampleCloud
            </a>
            <svg
              class="sidebarToggleLogo"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
            <a
              data-testid="login-link"
              href="/login"
            >
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn signInOut"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M192 277.4h189.7l-43.6 44.7L368 352l96-96-96-96-31 29.9 44.7 44.7H192v42.8z"
                      />
                      <path
                        d="M255.7 421.3c-44.1 0-85.5-17.2-116.7-48.4-31.2-31.2-48.3-72.7-48.3-116.9 0-44.1 17.2-85.7 48.3-116.9 31.2-31.2 72.6-48.4 116.7-48.4 44 0 85.3 17.1 116.5 48.2l30.3-30.3c-8.5-8.4-17.8-16.2-27.7-23.2C339.7 61 298.6 48 255.7 48 141.2 48 48 141.3 48 256s93.2 208 207.7 208c42.9 0 84-13 119-37.5 10-7 19.2-14.7 27.7-23.2l-30.2-30.2c-31.1 31.1-72.5 48.2-116.5 48.2zM448.004 256.847l-.849-.848.849-.849.848.849z"
                      />
                    </svg>
                  </div>
                  <span>
                    Log In
                  </span>
                </button>
              </div>
            </a>
          </nav>
        </div>
        <div
          class="pagesBlock"
        >
          <div
            class="root"
            data-testid="packs-page"
          >
            <div
              class="player"
            >
              <video
                autoplay=""
                class="playerVideo"
                loop=""
                src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
              />
              <div
                class="initialInformation"
              >
                <h1>
                  More than a sample library
                </h1>
                <p>
                  Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
                </p>
              </div>
            </div>
            <div
              class="inputContainer"
            >
              <label>
                Sounds
              </label>
              <div
                class="search"
              >
                <svg
                  class="searchIcon"
                  viewBox="0 0 512 512"
                  width="95"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
                <input
                  class="searchInput"
                  placeholder=""
                  type="text"
                  value=""
                />
                <svg
                  class="searchClose"
                  viewBox="0 0 352 512"
                  width="100"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  />
                </svg>
                <svg
                  class="searchDelete"
                  viewBox="0 0 640 512"
                  width="100"
                >
                  <path
                    d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="root"
            >
              <div
                class="loaderWrapper"
              >
                <div
                  class="loader"
                >
                  <div
                    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                    role="progressbar"
                    style="width: 40px; height: 40px;"
                  >
                    <svg
                      class="MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="20.2"
                        stroke-width="3.6"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="loaderPagination"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="packs-page"
    >
      <div
        class="player"
      >
        <video
          autoplay=""
          class="playerVideo"
          loop=""
          src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
        />
        <div
          class="initialInformation"
        >
          <h1>
            More than a sample library
          </h1>
          <p>
            Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
          </p>
        </div>
      </div>
      <div
        class="inputContainer"
      >
        <label>
          Sounds
        </label>
        <div
          class="search"
        >
          <svg
            class="searchIcon"
            viewBox="0 0 512 512"
            width="95"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            class="searchInput"
            placeholder=""
            type="text"
            value=""
          />
          <svg
            class="searchClose"
            viewBox="0 0 352 512"
            width="100"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
          <svg
            class="searchDelete"
            viewBox="0 0 640 512"
            width="100"
          >
            <path
              d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
            />
          </svg>
        </div>
      </div>
      <div
        class="root"
      >
        <div
          class="loaderWrapper"
        >
          <div
            class="loader"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="loaderPagination"
        />
      </div>
    </div>
    <div
      class="appContainer"
    >
      <div
        class="navbarBlock"
      >
        <nav
          class="navbar notAuth"
        >
          <a
            class="navbarLogo"
            data-testid="packs-link"
            href="/"
          >
            <div
              class="iconContainer"
            >
              <span
                aria-label="customer-service"
                class="anticon anticon-customer-service"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="customer-service"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 128c-212.1 0-384 171.9-384 384v360c0 13.3 10.7 24 24 24h184c35.3 0 64-28.7 64-64V624c0-35.3-28.7-64-64-64H200v-48c0-172.3 139.7-312 312-312s312 139.7 312 312v48H688c-35.3 0-64 28.7-64 64v208c0 35.3 28.7 64 64 64h184c13.3 0 24-10.7 24-24V512c0-212.1-171.9-384-384-384zM328 632v192H200V632h128zm496 192H696V632h128v192z"
                  />
                </svg>
              </span>
            </div>
            SampleCloud
          </a>
          <svg
            class="sidebarToggleLogo"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
          <a
            data-testid="login-link"
            href="/login"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn signInOut"
              >
                <div
                  class="iconContainer"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M192 277.4h189.7l-43.6 44.7L368 352l96-96-96-96-31 29.9 44.7 44.7H192v42.8z"
                    />
                    <path
                      d="M255.7 421.3c-44.1 0-85.5-17.2-116.7-48.4-31.2-31.2-48.3-72.7-48.3-116.9 0-44.1 17.2-85.7 48.3-116.9 31.2-31.2 72.6-48.4 116.7-48.4 44 0 85.3 17.1 116.5 48.2l30.3-30.3c-8.5-8.4-17.8-16.2-27.7-23.2C339.7 61 298.6 48 255.7 48 141.2 48 48 141.3 48 256s93.2 208 207.7 208c42.9 0 84-13 119-37.5 10-7 19.2-14.7 27.7-23.2l-30.2-30.2c-31.1 31.1-72.5 48.2-116.5 48.2zM448.004 256.847l-.849-.848.849-.849.848.849z"
                    />
                  </svg>
                </div>
                <span>
                  Log In
                </span>
              </button>
            </div>
          </a>
        </nav>
      </div>
      <div
        class="pagesBlock"
      >
        <div
          class="root"
          data-testid="packs-page"
        >
          <div
            class="player"
          >
            <video
              autoplay=""
              class="playerVideo"
              loop=""
              src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
            />
            <div
              class="initialInformation"
            >
              <h1>
                More than a sample library
              </h1>
              <p>
                Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
              </p>
            </div>
          </div>
          <div
            class="inputContainer"
          >
            <label>
              Sounds
            </label>
            <div
              class="search"
            >
              <svg
                class="searchIcon"
                viewBox="0 0 512 512"
                width="95"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                />
              </svg>
              <input
                class="searchInput"
                placeholder=""
                type="text"
                value=""
              />
              <svg
                class="searchClose"
                viewBox="0 0 352 512"
                width="100"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
              <svg
                class="searchDelete"
                viewBox="0 0 640 512"
                width="100"
              >
                <path
                  d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
                />
              </svg>
            </div>
          </div>
          <div
            class="root"
          >
            <div
              class="loaderWrapper"
            >
              <div
                class="loader"
              >
                <div
                  class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="loaderPagination"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
